internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,348 @@
+            on update cascade,
+        constraint scope_id_uq -- only allow one config per scope
+            unique(scope_id),
+    name wt_name,
+    constraint name_uq
+        unique(name, scope_id), 
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version
+);
+
+comment on table event_config is
+'event_config is a table where each row defines the event configuration for '
+'a scope.  Currently, the only supported scope is global';
+
+create trigger 
+  immutable_columns
+before
+update on event_config
+  for each row execute procedure immutable_columns(
+      'scope_id',
+      'create_time'
+      );
+
+create trigger
+    update_version_column
+after update on event_config
+    for each row execute procedure update_version_column();
+
+create trigger
+    update_time_column
+before update on event_config
+    for each row execute procedure update_time_column();
+
+create trigger
+    default_create_time_column
+before insert on event_config
+    for each row execute procedure default_create_time();
+
+create table event_type_enabled (
+    config_id wt_public_id
+        constraint event_config_fkey
+            references event_config(public_id)
+            on delete cascade
+            on update cascade,
+    event_type text not null
+        constraint event_type_enm_fkey
+            references event_type_enm(name)
+            on delete restrict
+            on update cascade,
+    primary key(config_id, event_type)
+);
+
+comment on table event_type_enabled is
+'event_type_enabled is a table where each row represents that eventing has '
+'been enabled for the specified event type in an event configuration';
+
+create table event_sink(
+    public_id wt_public_id primary key,
+    config_id wt_public_id not null
+        constraint event_config_fkey
+            references event_config(public_id)
+            on delete cascade
+            on update cascade
+);
+
+comment on table event_sink is 
+'event_sink is a table where each row represents a configured event sink';
+
+-- insert_event_sink_subtype() is a before insert trigger
+-- function for subtypes of event_sink
+create function insert_event_sink_subtype()
+returns trigger
+as $$
+begin
+insert into event_sink
+    (public_id, sink_id)
+values
+    (new.public_id, new.sink_id);
+return new;
+end;
+$$ language plpgsql;
+
+-- delete_event_sink_subtype() is an after delete trigger
+-- function for subtypes of event_sink
+create function delete_event_sink_subtype()
+returns trigger
+as $$
+begin
+delete from event_sink
+where public_id = old.public_id;
+return null; -- result is ignored since this is an after trigger
+end;
+$$ language plpgsql;
+
+create table event_file_sink(
+    public_id wt_public_id primary key,
+    sink_id wt_public_id not null
+        constraint event_sink_fkey
+            references event_sink(public_id)
+            on delete cascade
+            on update cascade,
```

Comment:
```suggestion
    config_id wt_public_id not null
        constraint event_config_fkey
            references event_config(public_id)
            on delete cascade
            on update cascade,
```
Examples: [base table](https://github.com/hashicorp/boundary/blob/248e1f36b5ed3ed24a0ec92e47a2de70fdb5b4ef/internal/db/schema/migrations/postgres/10/03_credential.up.sql#L6-L10) and [subtype table](https://github.com/hashicorp/boundary/blob/248e1f36b5ed3ed24a0ec92e47a2de70fdb5b4ef/internal/db/schema/migrations/postgres/10/04_vault_credential.up.sql#L5-L9)